src/dmd/dtemplate.d

```diff
@@ -8396,3 +8394,31 @@ void printTemplateStats()
         }
     }
 }
+
+/// Pair of MATCHes
+debug
+{
+    private struct MATCHpair
+    {
+        MATCH mta;  /// match template parameters by initial template arguments
+        MATCH mfa;  /// match template parameters by inferred template arguments
+
+        this(MATCH mta, MATCH mfa)
+        {
+            checkMATCH(mta);
+            checkMATCH(mfa);
+            this.mta = mta;
+            this.mfa = mfa;
+        }
+    }
+
+    void checkMATCH(MATCH m) { assert(m <= MATCH.max); }
+}
+else
+{
+    private struct MATCHpair
+    {
+        MATCH mta, mfa;
+    }
+}
```

Comment:
Instead of having duplicated definitions, isn't this a case for `invariant` ? Even if not, the `debug` should be inside the aggregate.

Reply:
yah, you're right